File: app/src/main/java/com/wineguesser/deductive/model/wine/WineDescriptors.java
package com.wineguesser.deductive.model.wine;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public abstract class WineDescriptors {

    private Clarity clarity;
    private Concentration concentration;
    private int rimVariation;
    private Staining staining;
    private int gasEvidence;
    private Intensity intensity;
    private Age age;
    private Oak oak;
    private Acidity acidity;
    private Alcohol alcohol;
    private Body body;
    private Texture texture;
    private int balanced;
    private Finish finish;
    private Complexity complexity;
    private Fault fault;
    private FruitCharacter fruitCharacter;
    private NonFruit nonFruit;
    private Earth earth;
    private Minerality minerality;
}

File: app/src/main/java/com/wineguesser/deductive/model/wine/WineDescriptors.kt
package com.wineguesser.deductive.model.wine

abstract class WineDescriptors(
    var clarity: Clarity? = null,
    var concentration: Concentration? = null,
    var rimVariation: Int = 0,
    var staining: Staining? = null,
    var gasEvidence: Int = 0,
    var intensity: Intensity? = null,
    var age: Age? = null,
    var oak: Oak? = null,
    var acidity: Acidity? = null,
    var alcohol: Alcohol? = null,
    var body: Body? = null,
    var texture: Texture? = null,
    var balanced: Int = 0,
    var finish: Finish? = null,
    var complexity: Complexity? = null,
    var fault: Fault? = null,
    var fruitCharacter: FruitCharacter? = null,
    var nonFruit: NonFruit? = null,
    var earth: Earth? = null,
    var minerality: Minerality? = null
)
